### ISSUE TITLE
关于 MoE aux_loss 中 top-k 维度归一化的疑问

### ISSUE THREAD

**Issue by wjj123a** [jingyaogong/minimind#797]
你好，我在阅读 MiniMind 的 MoE 实现时，注意到 MOEFeedForward.forward 中计算辅助负载均衡损失的代码是：

load = F.one_hot(topk_idx, self.config.num_experts).float().mean(0)
self.aux_loss = (load * scores.mean(0)).sum() * self.config.num_experts * self.config.router_aux_loss_coef

这里 topk_idx 的形状应该是 [num_tokens, num_experts_per_tok]，因此 F.one_hot(topk_idx, self.config.num_experts) 得到的形状是 [num_tokens, num_experts_per_tok, num_experts]。
当前代码使用 .mean(0) 后，load 的形状会变成 [num_experts_per_tok, num_experts]。当默认配置 num_experts_per_tok=1 时，这个问题不明显，因为只是多了一个长度为 1 的维度，后续和 scores.mean(0) 相乘时可以通过广播完成计算。
但如果将 num_experts_per_tok 设置为大于 1，例如 2 或更多，那么当前写法似乎会保留 top-k 维度，导致 aux loss 的尺度随 num_experts_per_tok 增大而变化。
想请教一下，这里是否应该同时对 token 维度和 top-k 维度求平均，例如：
load = F.one_hot(topk_idx, self.config.num_experts).float().mean(dim=(0, 1))
这样得到的 load 形状为 [num_experts]，语义上表示所有 token 的所有 expert assignment 中，每个 expert 被选中的平均比例，同时也可以让 aux loss 的尺度不随 num_experts_per_tok 改变。

**Comment by jingyaogong** [comment]
我对照了 `transformers` 的实现，HF的实现里 `qwen2_moe` / `qwen3_moe` 里的 [load_balancing_loss_func](cci:1://file:///Users/gongjy/project/ML/LLM/minimind/others/transformers/src/transformers/models/switch_transformers/modeling_switch_transformers.py:844:0-880:104) 也是保留 `top_k` 维度后再聚合，见： 

[modeling_qwen2_moe.py#L535-L614](https://github.com/huggingface/transformers/blob/main/src/transformers/models/qwen2_moe/modeling_qwen2_moe.py#L535-L614)
[modeling_qwen3_moe.py#L527-L606](https://github.com/huggingface/transformers/blob/main/src/transformers/models/qwen3_moe/modeling_qwen3_moe.py#L527-L606)

所以这里的写法和社区主流一致，也避免把 `num_experts_per_tok` 的变化强行折叠进 loss 标度里，这么做没有问题。
